### DISCUSSION TITLE
How to migrate Data from Firefox to Floorp?

### DISCUSSION THREAD

**Discussion by KAGEYAM4** [Floorp-Projects/Floorp#906] [Q&A]
Data ->
1. Extensions
2. Extensions Data
3. Tabs
4. Cookies
5. Settings ( not sure if this should be migrated or not )

**Comment by KAGEYAM4** [comment]
can i just copy paste the whole content from ~/.mozilla/firefox/<Random-number>/ to ~/.floorp/<Random-number>/ ?

**Reply by Telmesomething** [reply]
No, it's not possible for security reasons to copy everything in one go. However, it'll only take about ten minutes to migrate bookmarks and extensions as follows.

1. On Firefox, hit Ctrl + Shift + O to open the Library.
2. In the left hand pane double click Bookmarks Menu and you should see a list of them in the right hand pane.
3. On the top of the Library window click **Import and Backup** followed by **Export Bookmarks to HTML** and save the file somewhere.
4. Next, still in Firefox type: **about:profiles** in the URL bar.
5. In the Root directory of the Profile you want to copy extensions from click **Open Folder**.
6. Scroll down to the **extensions folder**, right click and copy and save it somewhere you can find it easily.
7. Close Firefox and open Floorp and open a new tab. 
8. Type **about:profiles** in the URL bar and hit Enter. 
9. In the Root directory click Open Folder, scroll down to the file called **places.sqlite** and delete it. _This will delete any bookmarks you have saved already_.
10. Hit Ctrl + Shift + O to open the Floorp Library. 
11. Click **Import and Backup** --> **Import Bookmarks from HTML** navigate to the **bookmarks.htm** file you created earlier and click Open. This will import your Firefox bookmarks into Floorp. It might take a couple of minutes to populate depending on how many there are.
12. Next, open a new Floorp tab and open the extensions folder you saved earlier. 
13. Drag & drop each extension that you want to keep into the new tab and confirm the prompt every time. It's laborious I know, but the only other way to do it is to go to the Mozilla site and download them that way. 

As far as the extensions folder is concerned most of them are just a set of alphanumeric names. To identify their real names open a blank tab in Firefox and type: **about:support** in the URL bar. Scroll down to the **Add-ons** paragraph and you'll find them all in there. 

I have configured Floorp to delete cache and cookies so there's no cookie file to move in my particular case, but the principal is the same as moving bookmarks to Floorp i.e. delete the cookie.sqlite file in Floorp first. 

cookie.sqlite and places.sqlite have the same names in the Profile directory, but other file extensions. You can ignore those.

**Reply by KAGEYAM4** [reply]
Thank you. This solves the issue of bookmark and extension ( partially ), for tabs i think i will use sideberyy and export tabs through them.

The current problem is extension data ( configs ) .

**Reply by KAGEYAM4** [reply]
i see `browser-extension-data` in firefox folder, not sure if it would work or not.

Does mainters reply in the `Discussion` or should i create a `Issue` ? Or discord ? 

I think having official documentation for migration to floorp or reverse would be good for new-users.

**Reply by Telmesomething** [reply]
`browser-extension-data` is just a set of folders all of which are empty on my system. 

Do the maintainers participate in discussions? Good question. I have seen a few, but not often. You can create a question by all means, but how long it will take to get a response is anybody's guess :)

**Reply by davidmarttila** [reply]
For extension settings, this has worked for me: https://www.reddit.com/r/firefox/comments/g4dk2c/tutorial_how_to_migrate_your_addons_and_their/?rdt=44096
